### PR TITLE
Object references

### PR DESCRIPTION
**Pull request by rowanseymour**
* Move object reference structs into their own package
* Switch `start_flow` and `set_preferred_channel` action to use reference objects for consistency with other actions

### CODE REVIEW

File: flows/actions/references.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	return &FlowReference{UUID: uuid, Name: name}
+}
+
+// FieldReference is a reference to field
+type FieldReference struct {
+	Key   flows.FieldKey `json:"key" validate:"required"`
+	Label string         `json:"label"`
+}
+
+// NewFieldReference creates a new field reference with the given UUID and key
```
Comment (nicpottier) [inline flows/actions/references.go]: comment

**Comment by nicpottier** [comment]
Do events never use these references?

**Comment by rowanseymour** [comment]
@nicpottier currently events just use the identifiers, e.g. the UUIDs or the field key.

**Comment by nicpottier** [comment]
Does that feel weird to you at all? I find it slightly odd that we treat these differently. Whats our argument for not being so?

**Comment by rowanseymour** [comment]
I really do see them as different - the references are part of the flow spec and I'm not sure events need to match that. For example:
* Group and label references in the spec can omit their UUID, and use an excellent template for their name value - none of those make sense in an event.
* A field reference will soon include it's data type - not something an event needs to be worried about.

**Comment by rowanseymour** [comment]
@nicpottier please re-review

**Comment by nicpottier** [comment]
Only bit I'm not positive about is `name_match` though I can't think of anything better.

**Pull request merged**